### PR TITLE
Update for Crystal v0.25

### PR DESCRIPTION
**Pull request by paulcsmith**
Still has some failing tests but it compiles. I need to read up on the JSON stuff to try to figure out why it isn't passing. I probably did something wrong.

Opening this now in case you want to take it over or have some ideas.

Thanks for an awesome lib :)

### CODE REVIEW

**Comment by paulcsmith** [comment]
Actually, it looks like the tests may be failing on master with v0.24 as well. So maybe this can be merged to get v0.25 support and then another PR for fixing the tests. For now, I'll use my forked branch in https://github.com/luckyframework/lucky_flow as it seems to work ok on v0.25 with this change :D

**Comment by ysbaddaden** [comment]
Is there no way to keep accessing the raw objects? I *really* don't like `JSON::Any` :cry: 

Also the patch is hard to read because of unrelated `crystal tool format` changes.

**Comment by bararchy** [comment]
@ysbaddaden everyone conforms to the tool format, maybe you should too ;)

File: src/webdriver.cr [hunk 1]
```diff
@@ -84,10 +84,10 @@ module Selenium
...
     private def failure(response)
       if response.headers["Content-Type"].starts_with?("application/json")
-        body = JSON.parse(response.body).raw.as(Hash)
-        status = body["status"].as(Int)
-        value = body["value"].as(Hash)
-        raise Selenium.error_class(status).new(value["message"].as(String))
+        body = JSON.parse(response.body)
+        status = body["status"].as_i
+        value = body["value"]
+        raise Selenium.error_class(status).new(body["value"]["message"].to_s)
```
Comment (ysbaddaden) [inline src/webdriver.cr]: You don't need `value = body["value"]` anymore and this it should be `body["value"]["message"].as_s`.

That's part of the reason I don't like `JSON::Type`, it introduces confusing namings: `.as_x` vs `.to_x`.

File: src/webdriver/session.cr [hunk 2]
```diff
@@ -29,8 +29,8 @@ module Selenium
       }
       response = driver.post("/session", body)
 
-      @id = response["sessionId"].as(String)
-      @capabilities = response["value"].as(Hash)
+      @id = response["sessionId"].to_s
```
Comment (ysbaddaden) [inline src/webdriver/session.cr]: `.as_s`

File: src/webdriver.cr (hunk 1 above)
Comment (paulcsmith) [reply src/webdriver.cr]: Yeah that is confusing. I'm not sure I understand the difference :S

**Comment by paulcsmith** [comment]
@ysbaddaden comments have ben addressed

File: src/webdriver.cr (hunk 1 above)
Comment (faustinoaq) [reply src/webdriver.cr]: > Yeah that is confusing. I'm not sure I understand the difference :S

- `as_x`: "type cast", type is already `x`
- `to_x`: "type covert", type is not `x` (commonly)

Comment (paulcsmith) [reply src/webdriver.cr]: That makes sense. Thank you :)

**Comment by ysbaddaden** [comment]
I released v0.4.0.

**Comment by paulcsmith** [comment]
Awesome! Thank you :)

**Pull request merged**